### DISCUSSION TITLE
nu and completions

### DISCUSSION THREAD

**Discussion by mock1328** [nushell/nushell#13971] [Q&A] [answered]
Is there a zsh-like completions support built-in nu, or it should be implemented as a 3rd-party plugin?
What I'm talking about, is that `nu` treats everything after command as a file argument, so when I type, e.g. `prime-run %some_app%`. I don't get completion for `%some_app%` part as I would have in `zsh`, instead `nu` provides me with filesystem completions. This works as it should for `sudo`, but I know a few commands that run the same way it does in `zsh`.
It's probably a different problem, but `nu` also doesn't seem to support command completions, am I right? E.g. trying to complete the `git pu...` wouldn't suggest me `pull` or `push`, it instead would provide filesystem completions again

**Comment by ghost** [comment, marked as answer]
<https://www.nushell.sh/book/custom_completions.html>

tl;dr: nu has only its own completions builtin. others have to be defined ([examples](https://github.com/nushell/nu_scripts/tree/main/custom-completions)) or provided by a external completer like carapace or fish (which you have to select in your config)
